### ISSUE TITLE
[Bug]: [benchmark][cluster] In the scenario of replica = 1, when scale out & in queryNode from 1 -> 3 -> 2, the number of segments increases

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#30977]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240229-dcfc3531-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):pulsar    
- SDK version(e.g. pymilvus v2.0.0rc2):2.4.0rc36
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task: multi-vector-scale-1-8s5x5

server:
```
NAME                                                              READY   STATUS             RESTARTS         AGE     IP              NODE         NOMINATED NODE   READINESS GATES
multi-vector-scale-1-8s5x5-etcd-0                                 1/1     Running            0                6h59m   10.104.21.233   4am-node24   <none>           <none>
multi-vector-scale-1-8s5x5-etcd-1                                 1/1     Running            0                6h59m   10.104.20.73    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-etcd-2                                 1/1     Running            0                6h59m   10.104.19.77    4am-node28   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-datacoord-57f88bd98c-4qgv6      1/1     Running            0                6h59m   10.104.30.159   4am-node38   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-datanode-76b8f774cb-wvwg2       1/1     Running            1 (6h54m ago)    6h59m   10.104.31.210   4am-node34   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-indexcoord-85b7944948-smjxw     1/1     Running            0                6h59m   10.104.31.211   4am-node34   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-indexnode-757c8b69f5-lllbv      1/1     Running            0                6h59m   10.104.4.207    4am-node11   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-proxy-59ffc56bfd-rdqkd          1/1     Running            1 (6h54m ago)    6h59m   10.104.30.160   4am-node38   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-querycoord-7cc756cdff-gcdwb     1/1     Running            1 (6h54m ago)    6h59m   10.104.31.209   4am-node34   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-ttttb      1/1     Running            0                6h59m   10.104.30.161   4am-node38   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-rootcoord-6c5f7d599-km6rr       1/1     Running            0                6h59m   10.104.30.158   4am-node38   <none>           <none>
multi-vector-scale-1-8s5x5-minio-0                                1/1     Running            0                6h59m   10.104.20.72    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-minio-1                                1/1     Running            0                6h59m   10.104.21.234   4am-node24   <none>           <none>
multi-vector-scale-1-8s5x5-minio-2                                1/1     Running            0                6h59m   10.104.19.75    4am-node28   <none>           <none>
multi-vector-scale-1-8s5x5-minio-3                                1/1     Running            0                6h59m   10.104.23.84    4am-node27   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-bookie-0                        1/1     Running            0                6h59m   10.104.20.74    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-bookie-1                        1/1     Running            0                6h59m   10.104.21.235   4am-node24   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-bookie-2                        1/1     Running            0                6h59m   10.104.23.85    4am-node27   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-bookie-init-wczvv               0/1     Completed          0                6h59m   10.104.20.63    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-broker-0                        1/1     Running            0                6h59m   10.104.5.55     4am-node12   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-proxy-0                         1/1     Running            0                6h59m   10.104.13.178   4am-node16   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-pulsar-init-k8fpq               0/1     Completed          0                6h59m   10.104.20.64    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-recovery-0                      1/1     Running            0                6h59m   10.104.5.56     4am-node12   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-zookeeper-0                     1/1     Running            0                6h59m   10.104.20.71    4am-node22   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-zookeeper-1                     1/1     Running            0                6h58m   10.104.25.199   4am-node30   <none>           <none>
multi-vector-scale-1-8s5x5-pulsar-zookeeper-2                     1/1     Running            0                6h57m   10.104.21.239   4am-node24   <none>           <none>
```

**scale out queryNode from 1 to 3 to 2**
```
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-ttttb      1/1     Running            0                6h59m   10.104.30.161   4am-node38   <none>           <none>
```
```
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-2r5nn      1/1     Running            0                 3m28s   10.104.24.40    4am-node29   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-4kpbw      1/1     Running            0                 3m28s   10.104.31.158   4am-node34   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-ttttb      1/1     Running            0                 7h2m    10.104.30.161   4am-node38   <none>           <none>
```
```
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-2r5nn      1/1     Running             0                 4h      10.104.24.40    4am-node29   <none>           <none>
multi-vector-scale-1-8s5x5-milvus-querynode-859fcfc669-ttttb      1/1     Running             0                 10h     10.104.30.161   4am-node38   <none>           <none>
```
<img width="1908" alt="截屏2024-03-01 16 52 01" src="https://github.com/milvus-io/milvus/assets/26307815/ff3d726f-fe15-4a5c-b461-789d298049cb">

**queryNode segment loaded num**
**The number of segments goes from 19 to 25 to 28, but there are actually 20**

<img width="1915" alt="截屏2024-03-01 16 56 19" src="https://github.com/milvus-io/milvus/assets/26307815/dbafa1f2-964c-408a-8932-1836aac0e561">
<img width="1912" alt="截屏2024-03-01 16 56 33" src="https://github.com/milvus-io/milvus/assets/26307815/148ce2c4-f600-4a84-81aa-9eea6b665a35">
<img width="1912" alt="截屏2024-03-01 16 56 46" src="https://github.com/milvus-io/milvus/assets/26307815/ba175578-68c6-4e1d-a369-519187b4d11f">

**queryNode Queryable Entity Num**
from 10M to 13.1M to 10.5M
<img width="1915" alt="截屏2024-03-01 17 01 01" src="https://github.com/milvus-io/milvus/assets/26307815/dba307b6-c370-44ba-b62d-249be9b3273f">
<img width="1918" alt="截屏2024-03-01 17 01 11" src="https://github.com/milvus-io/milvus/assets/26307815/a9ea14f7-68b2-4c5d-8a51-38c6e96e2024">
<img width="1910" alt="截屏2024-03-01 17 01 24" src="https://github.com/milvus-io/milvus/assets/26307815/e6fa63d3-58d9-44ff-a22f-7247a49f7ce6">

client log:
[client.log](https://github.com/milvus-io/milvus/files/14458671/client.log)

**the number of segment loaded is 20**
<img width="1919" alt="截屏2024-03-01 16 45 59" src="https://github.com/milvus-io/milvus/assets/26307815/0c3b86ab-0059-48b0-8fcc-c1372b9664d4">

![image](https://github.com/milvus-io/milvus/assets/26307815/849f0bed-692c-44c9-986f-e9a4111feeee)

### Steps To Reproduce

```markdown
1. create collection with fields:
                'float_vector': 128dim,
                'float_vector_1': 200dim,
                'binary_vector_1': 512dim,
                'array_varchar_1': max_length=10, max_capacity=5
                'int64_1': int64
                'id': int64, primary key
            2. build indexes:
                'float_vector': IVF_SQ8,
                'float_vector_1': HNSW,
                'binary_vector_1': BIN_IVF_FLAT,
                'int64_1': DEFAULT scalar index
                'id': INVERTED
            3. insert 10 million data
            4. flush collection
            5. build indexes again using the same params
            6. load collection replica=1
            7. concurrent request: <- scale queryNode from 1 to 3 to 2 during concurrent request
                - hybrid_search
```

### Anything else?

test result:
```
[2024-03-01 02:00:27,735 -  INFO - fouram]: Type     Name                                                                          # reqs      # fails |    Avg     Min     Max    Med |   req/s  failures/s (stats.py:789)
[2024-03-01 02:00:27,735 -  INFO - fouram]: --------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|----------- (stats.py:789)
[2024-03-01 02:00:27,735 -  INFO - fouram]: grpc     hybrid_search                                                                1378403     0(0.00%) |   3128     479   19895   2700 |   31.91        0.00 (stats.py:789)
[2024-03-01 02:00:27,735 -  INFO - fouram]: --------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|----------- (stats.py:789)
[2024-03-01 02:00:27,735 -  INFO - fouram]:          Aggregated                                                                   1378403     0(0.00%) |   3128     479   19895   2700 |   31.91        0.00 (stats.py:789)
[2024-03-01 02:00:27,735 -  INFO - fouram]:  (stats.py:790)
[2024-03-01 02:00:27,738 -  INFO - fouram]: [PerfTemplate] Report data: 
{'server': {'deploy_tool': 'helm',
            'deploy_mode': 'cluster',
            'config_name': 'cluster_2c2m',
            'config': {'queryNode': {'resources': {'limits': {'cpu': '8.0',
                                                              'memory': '32Gi'},
                                                   'requests': {'cpu': '5.0',
                                                                'memory': '17Gi'}},
                                     'replicas': 1},
                       'indexNode': {'resources': {'limits': {'cpu': '4.0',
                                                              'memory': '8Gi'},
                                                   'requests': {'cpu': '3.0',
                                                                'memory': '5Gi'}},
                                     'replicas': 1},
                       'dataNode': {'resources': {'limits': {'cpu': '2.0',
                                                             'memory': '2Gi'},
                                                  'requests': {'cpu': '2.0',
                                                               'memory': '2Gi'}}},
                       'cluster': {'enabled': True},
                       'pulsar': {},
                       'kafka': {},
                       'minio': {'metrics': {'podMonitor': {'enabled': True}}},
                       'etcd': {'metrics': {'enabled': True,
                                            'podMonitor': {'enabled': True}}},
                       'metrics': {'serviceMonitor': {'enabled': True}},
                       'log': {'level': 'debug'},
                       'image': {'all': {'repository': 'harbor.milvus.io/milvus/milvus',
                                         'tag': 'master-20240229-dcfc3531-amd64'}}},
            'host': 'multi-vector-scale-1-8s5x5-milvus.qa-milvus.svc.cluster.local',
            'port': '19530',
            'uri': ''},
 'client': {'test_case_type': 'ConcurrentClientBase',
            'test_case_name': 'test_concurrent_locust_ivf_sq8_hybrid_search_cluster',
            'test_case_params': {'dataset_params': {'metric_type': 'L2',
                                                    'dim': 128,
                                                    'scalars_index': {'int64_1': {},
                                                                      'id': {'index_type': 'INVERTED'}},
                                                    'vectors_index': {'float_vector_1': {'index_type': 'HNSW',
                                                                                         'index_param': {'M': 8,
                                                                                                         'efConstruction': 200},
                                                                                         'metric_type': 'L2'},
                                                                      'binary_vector_1': {'index_type': 'BIN_IVF_FLAT',
                                                                                          'index_param': {'nlist': 2048},
                                                                                          'metric_type': 'JACCARD'}},
                                                    'scalars_params': {'float_vector_1': {'params': {'dim': 200},
                                                                                          'other_params': {'dataset': 'text2img',
                                                                                                           'dim': 200}},
                                                                       'binary_vector_1': {'params': {'dim': 512},
                                                                                           'other_params': {'dataset': 'binary',
                                                                                                            'dim': 512}},
                                                                       'array_varchar_1': {'params': {'max_length': 10,
                                                                                                      'max_capacity': 5},
                                                                                           'other_params': {'varchar_filled': False}}},
                                                    'dataset_name': 'sift',
                                                    'dataset_size': '10m',
                                                    'ni_per': 5000},
                                 'collection_params': {'other_fields': ['float_vector_1',
                                                                        'array_varchar_1',
                                                                        'int64_1',
                                                                        'binary_vector_1'],
                                                       'shards_num': 2},
                                 'load_params': {'replica_number': 1},
                                 'index_params': {'index_type': 'IVF_SQ8',
                                                  'index_param': {'nlist': 1024}},
                                 'concurrent_params': {'concurrent_number': 100,
                                                       'during_time': '12h',
                                                       'interval': 20,
                                                       'spawn_rate': None},
                                 'concurrent_tasks': [{'type': 'hybrid_search',
                                                       'weight': 1,
                                                       'params': {'nq': 1,
                                                                  'top_k': 10,
                                                                  'reqs': [{'search_param': {'nprobe': 32},
                                                                            'anns_field': 'float_vector',
                                                                            'expr': 'int64_1 '
                                                                                    '< '
                                                                                    '100000'},
                                                                           {'search_param': {'ef': 64},
                                                                            'anns_field': 'float_vector_1',
                                                                            'expr': 'id '
                                                                                    '> '
                                                                                    '10',
                                                                            'top_k': 60},
                                                                           {'search_param': {'nprobe': 64},
                                                                            'anns_field': 'binary_vector_1',
                                                                            'top_k': 2000}],
                                                                  'rerank': {'WeightedRanker': [0.3,
                                                                                                0.4,
                                                                                                0.3]},
                                                                  'timeout': 60,
                                                                  'random_data': True}}]},
            'run_id': 2024022942028850,
            'datetime': '2024-02-29 10:56:42.710178',
            'client_version': '2.4.0'},
 'result': {'test_result': {'index': {'RT': 3456.4124,
                                      'float_vector_1': {'RT': 2993.9311},
                                      'binary_vector_1': {'RT': 2477.0517},
                                      'int64_1': {'RT': 539.078},
                                      'id': {'RT': 2.025}},
                            'insert': {'total_time': 803.5029,
                                       'VPS': 12445.5058,
                                       'batch_time': 0.4018,
                                       'batch': 5000},
                            'flush': {'RT': 3.0527},
                            'load': {'RT': 24.6291},
                            'Locust': {'Aggregated': {'Requests': 1378403,
                                                      'Fails': 0,
                                                      'RPS': 31.91,
                                                      'fail_s': 0.0,
                                                      'RT_max': 19895.49,
                                                      'RT_avg': 3128.03,
                                                      'TP50': 2700.0,
                                                      'TP99': 14000.0},
                                       'hybrid_search': {'Requests': 1378403,
                                                         'Fails': 0,
                                                         'RPS': 31.91,
                                                         'fail_s': 0.0,
                                                         'RT_max': 19895.49,
                                                         'RT_avg': 3128.03,
                                                         'TP50': 2700.0,
                                                         'TP99': 14000.0}}}}} 
```

**Comment by weiliu1031** [comment]
known issue, sealed segment leak, should be fixed by #31040 , please verify this.

**Comment by wangting0128** [comment]
### verification passed

image：master-20240315-b7247531
